drivers/SmartThings/zigbee-switch/src/tuya-multi/init.lua

```diff
@@ -0,0 +1,123 @@
+    if string.sub(device:get_manufacturer(),1,3) == TUYA_MFR_HEADER then  -- if it is a tuya device, then send the magic packet
+        local subdriver = require("tuya-multi")
+        return true, subdriver
+    end
+    return false
+end
+
+local function send_discover_command(device)
+    local discover_command_payload = "\x00\x00\xff"      -- Start Attribute: 0x0000, Maximum Number: 255
+    local header_args = {
+      cmd = data_types.ZCLCommandId(DISCOVER_ATTR_RST_ID)
+    }
+    local zclh = zcl_messages.ZclHeader(header_args)
+    local addrh = messages.AddressHeader(
+      zb_const.HUB.ADDR,
+      zb_const.HUB.ENDPOINT,
+      device:get_short_address(),
+      device:get_endpoint(clusters.Basic.ID),
+      zb_const.HA_PROFILE_ID,
+      clusters.Basic.ID
+    )
+    local message_body = zcl_messages.ZclMessageBody({
+      zcl_header = zclh,
+      zcl_body = generic_body.GenericBody(discover_command_payload)
+    })
+    local send_message = messages.ZigbeeMessageTx({
+        address_header = addrh,
+        body = message_body
+    })
+    device:send(send_message)
+end
+
+local function read_attribute_function(device, cluster_id, attr_id)
+    local read_body = read_attribute.ReadAttribute( attr_id )
+    local zclh = zcl_messages.ZclHeader({
+      cmd = data_types.ZCLCommandId(read_attribute.ReadAttribute.ID)
+    })
+    local addrh = messages.AddressHeader(
+        zb_const.HUB.ADDR,
+        zb_const.HUB.ENDPOINT,
+        device:get_short_address(),
+        device:get_endpoint(cluster_id.value),
+        zb_const.HA_PROFILE_ID,
+        cluster_id.value
+    )
+    local message_body = zcl_messages.ZclMessageBody({
+      zcl_header = zclh,
+      zcl_body = read_body
+    })
+    return messages.ZigbeeMessageTx({
+      address_header = addrh,
+      body = message_body
+    })
+end
+
+local function do_payload(rx)  -- use discover attribute response payload to do read attribute payload
+    local out_str = " "
+    for i = 1, #rx.body_bytes do
+      out_str = out_str .. string.format(" %02X", string.byte(rx.body_bytes:sub(i,i)))
+    end
+    local bytes = {}
+    for b in out_str:gmatch("%S+") do
+        table.insert(bytes, b)
+    end
+    table.remove(bytes, 1) -- remove first type which means discover attribute command is completed
+    local result = {}
+    for i = 1, #bytes, 3 do   -- remove data type information
+        local low = bytes[i]
+        local high = bytes[i+1]
+        if high then
+            local value = tonumber(high, 16) * 0x100 + tonumber(low, 16)
+            table.insert(result, value)
+        end
+    end
+    return result
+end
+
+local do_configure = function(self, device)
+    device:refresh()
+    device:configure()
+    send_discover_command(device)
+end
+
+local function discover_response_message_handler(driver, device, zb_rx)
+    if zb_rx.address_header.cluster.value == clusters.Basic.ID and zb_rx.body.zcl_header.cmd.value == DISCOVER_ATTR_RSP_ID then 
```

Comment:
If this function is called, both of these have already been validated, so you don't need to check again here.